pkg/ddl/ddl_worker.go

```diff
@@ -899,7 +907,27 @@ func (w *worker) prepareTxn(job *model.Job) (kv.Transaction, error) {
 	return txn, err
 }
 
-func (w *worker) HandleDDLJobTable(d *ddlCtx, job *model.Job) (int64, error) {
+// runOneJobStep runs one step of the DDL job and persist the states change. One
+// *step* is defined as the following reason:
```

Comment:
other *step*: 1. reorg also has it's own state change, 2. reorg runs in async, it will exter/exist this function to check whether the async routine is done, there is no state change during this time

Reply:
CI shows the *step* `onLockTables` is not a schema state change. LOCK TABLES may have many table arguments and each step update one table's lock state and persist it. I'll come up with a better comment tomorrow, and align the code to fix UT 😂 Maybe I should keep the old `needUpdateRawArgs` behaviour (if no `runErr` we should marshal RawArgs), and fix wrong `runErr` nilness to close https://github.com/pingcap/tidb/issues/54687

Reply:
> reorg runs in async, it will exter/exist this function to check whether the async routine is done, there is no state change during this time

please add this one too

Reply:
updated some function comments. https://github.com/pingcap/tidb/pull/54685/commits/f782a5208216214e8ca4e9b30872e7584b6859ac

reorg is a bit complex as a breif example in function comments, I only added onLockTables. Please check if it's clear enough.